1.Core/UIElements/Web/TextField.cs

```diff
@@ -7,9 +10,43 @@ public class TextField : BaseWebElement, ITypeable
     {
         public TextField(string name, Locator locator) : base(name, locator) { }
 
+        public void Clear()
+        {
+            try
+            {
+                GenericWebDriver.Wait.Until(ExpectedConditions.ElementIsVisible(Locator.GetBy()));
```

Comment:
It is not necesary to wait for the element to be visible if you are waiting for it to be clickable.
You can confirm that by going to the [source code](https://github.com/DotNetSeleniumTools/DotNetSeleniumExtras/blob/master/src/WaitHelpers/ExpectedConditions.cs), check it out. ElementToBeClickable already checks if the element is visible.

Reply:
Cool, I didn't know that thanks. Fixed.